### ISSUE TITLE
Taging via configuration doesn't work for BAN requests

### ISSUE THREAD

**Issue by tomaszhanc** [FriendsOfSymfony/FOSHttpCacheBundle#241]
I add tags via YAML file, with configuration similiar to this one:

``` yml
    fos_http_cache:
        tags:
            rules:
                -
                    match:
                        path: ^/config
                    tag: [config]
```

It works great for responses, but it doesn't add these tags for BAN request to Varnish after POST method which is returning a status of 201. I checked adding tags via annotation and it works ok there. I've found out that these lines in RuleMatcher are causing the problem:

``` php
    $status = array(200, 203, 204, 300, 301, 302, 404, 410);
    if (!empty($this->criteria['additional_cacheable_status'])) {
        $status = array_merge($status, $this->criteria['additional_cacheable_status']);
    }
    if (!in_array($response->getStatusCode(), $status)) {
        return false;
    }
```

Adding to confgiuration option `additional_cacheable_status: [201]` solves the problem, but it's a bit tedious to do for every rule. Can we add this status code as a default?

**Comment by dbu** [comment]
hm, looks like cacheable status is not exactly the same as invalidatable status... we should not cache responses with status 201 but invalidating sounds right.

if you see an idea how to distinguish the two things without making everything very complicated, please do a pull request against master.

**Comment by teohhanhui** [comment]
#230 only fixed it for `invalidation` rules (config). See #69. It seems to be a longstanding issue...

**Comment by dbu** [comment]
yeah nobody got around to implement #69 so far, unfortunately. if anybody has time to contribute that, would be awesome!

**Comment by ddeboer** [comment]
Fixed in #354.
